sources/api/netdog/src/networkd_status.rs

```diff
@@ -0,0 +1,285 @@
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub(crate) struct NetworkDInterfaceStatus {
+    pub(crate) name: InterfaceName,
+    #[serde(rename = "DNS")]
+    pub(crate) dns: Option<Vec<NetworkDDnsConfig>>,
+    pub(crate) search_domains: Option<Vec<SearchDomain>>,
+    #[serde(rename = "Addresses", deserialize_with = "from_networkctl_addresses")]
+    pub(crate) addresses: Vec<IpAddr>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct NetworkDDnsConfig {
+    #[serde(rename = "Address", deserialize_with = "ipaddr_from_vec_de")]
+    pub(crate) address: IpAddr,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct SearchDomain {
+    #[serde(rename = "Domain")]
+    pub(crate) domain: String,
+}
+
+// get an IpAddr from a Vec<u8> (could be 4 or 16 length)
+fn ipaddr_from_vec(address_vec: Vec<u8>) -> Result<IpAddr> {
+    match address_vec.len() {
+        // Already checked that its exactly 4 u8 long
+        4 => Ok(IpAddr::from(
+            TryInto::<[u8; 4]>::try_into(address_vec).expect("4 bytes"),
+        )),
+        // Already checked that its exactly 16 u8 long
+        16 => Ok(IpAddr::from(
+            TryInto::<[u8; 16]>::try_into(address_vec).expect("16 bytes"),
+        )),
+        _ => error::BadIpAddressSnafu {
+            input: address_vec,
+            msg: "invalid length, must be 4 or 16 octets".to_string(),
+        }
+        .fail(),
+    }
+}
+
+fn ipaddr_from_vec_de<'de, D>(deserializer: D) -> std::result::Result<IpAddr, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let address_vec: Vec<u8> = Deserialize::deserialize(deserializer)?;
+    ipaddr_from_vec(address_vec).map_err(D::Error::custom)
+}
+
+fn from_networkctl_addresses<'de, D>(deserializer: D) -> std::result::Result<Vec<IpAddr>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Debug, Deserialize)]
+    #[serde(rename_all = "PascalCase")]
+    struct NetworkctlAddress {
+        address: Vec<u8>,
+    }
+    let addresses: Vec<NetworkctlAddress> = Deserialize::deserialize(deserializer)?;
+    let mut addrs = Vec::new();
+    for addr in addresses.iter() {
+        addrs.push(ipaddr_from_vec(addr.address.clone()).map_err(D::Error::custom)?);
+    }
+    Ok(addrs)
+}
+
+impl NetworkDInterfaceStatus {
+    pub(crate) fn new(link: String) -> Result<Self> {
+        let systemd_networkctl_result = Command::new(NETWORKCTL)
+            .arg("status")
+            .arg("--json=pretty")
+            .arg(link)
+            .output()
+            .context(error::NetworkctlExecutionSnafu)?;
+        ensure!(
+            systemd_networkctl_result.status.success(),
+            error::FailedNetworkctlSnafu {
+                stderr: String::from_utf8_lossy(&systemd_networkctl_result.stderr)
+            }
+        );
+
+        let status_output = String::from_utf8(systemd_networkctl_result.stdout)
+            .context(error::NetworkctlParsingSnafu {})?;
+
+        let networkd_status =
```

Comment:
Nit: Rather than converting the command output to a string first, we could skip a step and use `serde_json.from_slice()`.  It'd be something like:
```
let networkd_status = serde_json::from_slice(&networkctl_result.stdout).context(...)
```